autometa/config/databases.py

```diff
@@ -111,185 +232,388 @@ def format_nr(self):
 
         """
         db_infpath = self.config.get('ncbi', 'nr')
+        db_infpath_md5 = f'{db_infpath}.md5'
         db_outfpath = db_infpath.replace('.gz', '.dmnd')
-        if not self.dryrun and not os.path.exists(db_outfpath):
-            diamond.makedatabase(
-                fasta=nr, database=db_infpath, nproc=self.nproc)
+        checksums_match = False
+        if os.path.exists(db_infpath_md5) and os.path.exists(db_outfpath):
+            # If nr.dmnd.md5 exists, then we will check if nr.gz.md5 is matching
+            current_checksum = read_checksum(db_infpath_md5)
+            current_hash, __ = current_checksum.split()
+            host = self.config.get('ncbi', 'host')
+            remote_checksum = self.get_remote_checksum(host, 'nr')
+            remote_hash, __ = remote_checksum.split()
+            if current_hash == remote_hash:
+                logger.debug(f'nr checksums match, skipping...')
+                checksums_match = True
+        dmnd_md5 = f'{db_outfpath}.md5'
+        do_not_upgrade = os.path.exists(db_outfpath) and not self.upgrade
+        if self.dryrun or do_not_upgrade or checksums_match:
+            self.config.set('ncbi', 'nr', db_outfpath)
+            logger.debug(f'{dmnd_md5} exists: {os.path.exists(dmnd_md5)}')
+            logger.debug(f'set ncbi nr: {db_outfpath}')
+            return
+        diamond.makedatabase(
+            fasta=db_infpath, database=db_outfpath, nproc=self.nproc)
+        # Write checksum for nr.dmnd
+        write_checksum(db_outfpath, dmnd_md5)
+
+        if os.path.basename(db_infpath) == 'nr.gz':
+            # nr.gz will be removed after successful nr.dmnd construction
+            os.remove(db_infpath)
+
         self.config.set('ncbi', 'nr', db_outfpath)
         logger.debug(f'set ncbi nr: {db_outfpath}')
 
     def extract_taxdump(self):
-        """Extract autometa required files from ncbi taxdump directory.
+        """Extract autometa required files from ncbi taxdump.tar.gz archive
+        into ncbi databases directory and update user config with extracted paths.
+
+        This only extracts nodes.dmp, names.dmp and merged.dmp from
+        taxdump.tar.gz if the files do not already exist. If `upgrade`
+        was originally supplied as `True` to the Databases instance, then the
+        previous files will be replaced by the new taxdump files.
 
-        Extracts nodes.dmp, names.dmp and merged.dmp from taxdump.tar.gz
+        After successful extraction of the files, a checksum will be written of the
+        archive for future checking and then the archive will be removed to save user disk space.
```

Comment:
I think this is wrong - I couldn't see where the .tar file is deleted in this function, so the comment should be changed. For what it is worth, I think we should leave the parent .tar.gz file on the disk.

Reply:
I've removed the lines that discard `taxdump.tar.gz`. This routine was at the end of the function. Lines 307-309